convert2rhel/subscription.py

```diff
@@ -385,6 +394,26 @@ def __call__(self):
             register_server.Stop(i18n.SUBSCRIPTION_MANAGER_LOCALE, dbus_interface="com.redhat.RHSM1.RegisterServer")
 
 
+def parse_url(server_url):
+    """Parse a string as a url in a manner mostly compatible with subscription-manager."""
+    # This is an adaptation of what subscription-manager's cli enforces:
+    # https://github.com/candlepin/subscription-manager/blob/main/src/rhsm/utils.py#L112
+
+    # It should look like it has a scheme followed by other information.
+    if not re.match("^.+://.+", server_url):
```

Comment:
i'd use `[^:]+` instead of `.+`, to avoid potential greedyness of regexps

Reply:
Good idea.

Reply:
change pushed in a new commit.